### DISCUSSION TITLE
How to use inside Subflow

### DISCUSSION THREAD

**Discussion by Stasfe** [GogoVega/node-red-contrib-firebase-realtime-database#65] [Q&A] [answered]
Hey I can not figure out how I can use this node inside the subflow with enviroment variables. I want to have my path like this: /4100/${path}/status but I get error because I am not allowed to use $ sign. 
I was trying around with msg.topic to set the path. Anyways the node need a path by itself, I can not leave it empty. 
here are some screenshots of what I've been trying:
with funktion node:
![image](https://github.com/GogoVega/node-red-contrib-firebase-realtime-database/assets/46917944/f7a88010-e26b-4fea-b9d1-92697e148c75)

with change node:
![image](https://github.com/GogoVega/node-red-contrib-firebase-realtime-database/assets/46917944/877af721-78be-496f-8918-42f47424d914)

also tried something like this: 

![image](https://github.com/GogoVega/node-red-contrib-firebase-realtime-database/assets/46917944/fe29ac00-dce4-4245-9a55-d85626e57972)

still getting only the path updated wich is in the node itself.

**Comment by GogoVega** [comment, marked as answer]
Hi,
To use environment variables, you must select this mode in the field types.

Below is an example of the node you want:

```json
[{"id":"422d4a3c06deb40c","type":"change","z":"15cdbf1d258ebb57","name":"","rules":[{"t":"set","p":"topic","pt":"msg","to":"/4100/${path}/status","tot":"env"}],"action":"","property":"","from":"","to":"","reg":false,"x":320,"y":600,"wires":[["ddafd3839d415f9d"]]}]
```

Currently version 0.5 does not yet support `env` as a path type, I will add it for v0.6 😉 

I hope I answered your question.

**Comment by Stasfe** [comment]
very nice thank you

**Comment by GogoVega** [comment]
Hi @Stasfe,
If you wish, v0.6 is available in beta. You can install it directly from the Palette Manager.
After restarting Node-RED, you will have a migration wizard to migrate your configs.
Then enjoy 😉

**Reply by Stasfe** [reply]
Hey, I had some troubles installing the new Version, because I had to update the whole enviroment. But now its done and works pretty nice. I like it. 
I have a question about Firebase in node. In the path section its only possible to set up a String? 
![image](https://github.com/user-attachments/assets/698a869a-a824-4fc1-a8ff-8801534a7a1e)

I have like 24 such pathes that I need to listen to, so my plan was to create a subflow, so I can use it for multiple node red instances. Is there a work around to get env varible here?

**Reply by GogoVega** [reply]
Hi,
Yep, since the `env` field type can support the `msg` expression, you need to select the dynamic listener (`msg.listener`).

This mode enables dynamic options for the path and adds an input to the node. Don't forget to send `msg.listener` because in this mode it's no longer defined in the node config.

**Reply by GogoVega** [reply]
Hi @Stasfe,
I will in a few days modify the `Firebase In` node so that the `Path` field displays the full options by default. And depending on the option chosen, it will add or not an input to the node. With this way, you will have the possibility to statically define the listener type and dynamically the path (and vice versa).

**Reply by Stasfe** [reply]
sounds awesome. love this node, works very well.

**Reply by GogoVega** [reply]
Available now in v0.7.3 🎉 ( with some other enhancements 😉🤫 )

**Reply by Stasfe** [reply]
So I have tryed the firebase in node again inside my subflow. As I see I cannot listen to a value with env variable. As you mention before, I have to use msg.listener. How it exactly works? Do I have to trigger? I need (would be nice) firebase in node inside subflow to listen in realtime for env variable. So i would be able to use many instances of subflow and just change the variable. 
Right now I use some workaround.

**Reply by GogoVega** [reply]
Your previous post asked if the path could use env var, that's what was added:

<img width="448" alt="Capture d’écran 2025-03-05 à 11 09 28" src="https://github.com/user-attachments/assets/5f062244-a9dc-4a3f-895c-337f72f755a4" />

```json
[{"id":"7d6549ddff974191","type":"firebase-in","z":"15cdbf1d258ebb57","name":"","constraints":{},"database":"e8796a1869e179bc","inputs":0,"listenerType":"value","outputType":"auto","passThrough":false,"path":"test","pathType":"env","useConstraints":false,"x":330,"y":200,"wires":[[]]}]
```

You don't see the same thing?

**Comment by Stasfe** [comment]
I reinstalled now and see it now. Thank you for quick reply

**Reply by GogoVega** [reply]
Good news 🙂 weird to reinstall it, I'd say it's more of a cache issue but whatever.
